rcljava/src/main/java/org/ros2/rcljava/node/NodeImpl.java

```diff
@@ -392,103 +401,299 @@ public final String getName() {
     return this.name;
   }
 
+  public void declareParameter(ParameterVariant parameter) {
+    declareParameter(parameter, new rcl_interfaces.msg.ParameterDescriptor().setName(parameter.getName()));
+  }
+
+  private void internalDeclareParameter(ParameterVariant parameter, rcl_interfaces.msg.ParameterDescriptor descriptor) {
+    if (this.parameters.containsKey(parameter.getName())) {
+      throw new IllegalArgumentException("Parameter is already declared");
+    }
+
+    ParameterAndDescriptor pandd = new ParameterAndDescriptor();
+    pandd.parameter = parameter;
+    pandd.descriptor = descriptor;
+
+    this.parameters.put(parameter.getName(), pandd);
+  }
+
+  public void declareParameter(ParameterVariant parameter, rcl_interfaces.msg.ParameterDescriptor descriptor) {
+    synchronized (parametersMutex) {
+      internalDeclareParameter(parameter, descriptor);
+    }
+  }
+
+  public void declareParameters(List<ParameterVariant> parameters, List<rcl_interfaces.msg.ParameterDescriptor> descriptors) {
+    if (parameters.size() != descriptors.size()) {
+      throw new IllegalArgumentException("Parameters length must be equal to descriptors length");
+    }
+    Iterator<ParameterVariant> itp = parameters.iterator();
+    Iterator<rcl_interfaces.msg.ParameterDescriptor> itpd = descriptors.iterator();
+    synchronized (parametersMutex) {
```

Comment:
Following my previous comment, this could become
```suggestion
    synchronized (this) {
```